modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageFlusher.java

```diff
@@ -446,20 +444,63 @@ public int sendMessagesToSubscriptions(String destination, Set<AndesMessageMetad
                         //if we continue message order will break
                         break;
                     }
-                } else { //normal topic message
-                    if (numOfCurrentMsgDeliverySchedules == subscriptions4Queue.size()) {
+                } else {
```

Comment:
Any Possibility of breaking this long method in two? lots of branching makes the maintainability/readability of the code a problem.